### ISSUE TITLE
Remove excess timestamp flexibility in stratum / getblocktemplate protocols

### ISSUE THREAD

**Issue by feeleep75** [zcash/zcash#1880]
Hi,

I noticed that some blocks have strange timestamp: my pool discovered a block 12401, work has been submitted to daemon at 2016/11/18 09:44:01 CET, however block timestamp shows 2016-11-18 09:14:40 CET...

**Comment by daira** [comment]
When you say "my pool", do you mean that you're a user of that pool, or an operator?

**Comment by feeleep75** [comment]
Operator but what's the difference?

**Comment by daira** [comment]
Just in how much you can be expected to know about the pool setup and what diagnostics you're able to perform.

What software is your pool running, and how did you set it up?

**Comment by sammy007** [comment]
@daira everyone is running their own software, because there is no FOSS pool. I guess the problem is user allowed to edit `curtime`.

```
{"id": 4, "method": "mining.submit", "params": ["WORKER_NAME", "JOB_ID", "TIME", "NONCE_2", "EQUIHASH_SOLUTION"]}
```

I believe it's a good practice to ignore `TIME` param from miner and force server-side curtime. I hope it will be flagged in stratum ZIP. There is no practical reason to allow miner to roll their own timestamp.

**Comment by ioptio** [comment]
Doing some housekeeping. @feeleep75 is this still and issue for you? If so, can you respond to the comments above to help us understand?

**Comment by feeleep75** [comment]
This was never my problem :) - i was reporting an issue that some of the pools/miners are mining blocks back in time :)

**Comment by ioptio** [comment]
Are you still seeing this behavior?

**Comment by ioptio** [comment]
Closing this issue, please refer to #1889 for recent discussions

**Comment by nathan-at-least** [comment]
Hi, I'm re-opening this issue, and I've retitled it to follow @sammy007 's [proposal above](https://github.com/zcash/zcash/issues/1880#issuecomment-261757954). I'm not familiar enough with the mining-pool protocols to know how feasible it is to remove client-supplied timestamps. I'll ping @str4d who has been working on a stratum spec to see if they consider this a possible/valuable change.

BTW, @feeleep75, you're right that block timestamps can be before local time (or even previous blocks in some frequent cases). I just filed #2022 to make it easier to analyze that behavior from the RPC interface, and we have #1889 that's a proposed longer-term protocol fix.
